test_reproducibility.py

```diff
@@ -0,0 +1,41 @@
+from pathlib import Path
```

Comment:
Use a dedicated testing framework (`pytest` is strongly preferred)  instead of a regular script.

Reply:
but pytest will handle this function and i did not use pytest here by intention ;)

Reply:
I just assumed its not meant for pytest because it doesn't quite look like it is. With pytest we could take advantage of multiple mechanisms:

1. Fixtures instead of constats
2. Parametrizing the test functions
3. Using pytest tempfile instead of pythons

Then you can also skip the last if `__name__ == "__main__"` block